ports/esp32/machine_touchpad.c

```diff
@@ -102,14 +128,53 @@ static mp_obj_t mtp_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_
 
     static int initialized = 0;
     if (!initialized) {
+    #if SOC_TOUCH_SENSOR_VERSION == 1 || SOC_TOUCH_SENSOR_VERSION == 2
         touch_pad_init();
         touch_pad_set_fsm_mode(TOUCH_FSM_MODE_TIMER);
+    #elif SOC_TOUCH_SENSOR_VERSION == 3
+    	touch_sensor_sample_config_t sample_cfg[3] = {
+		TOUCH_SENSOR_V3_DEFAULT_SAMPLE_CONFIG(1, 1, 1),
+        	TOUCH_SENSOR_V3_DEFAULT_SAMPLE_CONFIG(2, 1, 1),
+        	TOUCH_SENSOR_V3_DEFAULT_SAMPLE_CONFIG(4, 1, 1),
+	};
+	touch_sensor_config_t sens_cfg = TOUCH_SENSOR_DEFAULT_BASIC_CONFIG(3, sample_cfg);
+        esp_err_t err = touch_sensor_new_controller(&sens_cfg, &s_sens_handle);
+	if (err != ESP_OK) {
+    	   mp_raise_ValueError(MP_ERROR_TEXT("Touch pad error 1"));
```

Comment:
I can see why you've numbered these (`1`, `2`, etc) but it's not something that we do in MicroPython.

Suggest either making the errors the same or add some better description of what failed. Most of these errors should never happen unless there's an internal driver bug, yes?